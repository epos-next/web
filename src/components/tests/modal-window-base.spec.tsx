import React from "react";
import ModalWindowBase from "@components/modal-window-base";
import { fireEvent, render } from "@testing-library/react";

it("should be open", () => {
    const { asFragment } = render(<ModalWindowBase isOpen={true} onClose={jest.fn} />)
    expect(asFragment()).toMatchSnapshot()
});

it("should be closed", () => {
    const { asFragment } = render(<ModalWindowBase isOpen={false} onClose={jest.fn} />)
    expect(asFragment()).toMatchSnapshot()
});

it("should call callback if user click on close button", () => {
    const callback = jest.fn()
    const { getByAltText } = render(<ModalWindowBase isOpen={false} onClose={callback} />)
    fireEvent.click(getByAltText("close"))
    expect(callback).toHaveBeenCalledTimes(1)
});
