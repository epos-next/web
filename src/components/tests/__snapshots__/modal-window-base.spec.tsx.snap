// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should be closed 1`] = `
<DocumentFragment>
  .c3 {
  position: absolute;
  right: 35px;
  top: 35px;
  cursor: pointer;
}

.c2 {
  width: 100%;
  max-width: 822px;
  background-color: white;
  border-radius: 25px;
  padding: 35px;
  position: relative;
  -webkit-transition: 400ms cubic-bezier(0.83,0,0.17,1);
  transition: 400ms cubic-bezier(0.83,0,0.17,1);
}

.c0 {
  width: 100vw;
  height: 100vh;
  z-index: 100000000;
  background-color: rgba(0,0,0,0.35);
  position: fixed;
  top: 0;
  left: 0;
  padding: 20px;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-transition: opacity 400ms ease,background-color 400ms;
  transition: opacity 400ms ease,background-color 400ms;
}

.c0[data-open=false] {
  pointer-events: none;
  opacity: 0;
}

.c0[data-open=false] .c1 {
  -webkit-transform: translateY(300px);
  -ms-transform: translateY(300px);
  transform: translateY(300px);
  opacity: 0;
}

@media screen and (max-width:768px) {
  .c3 {
    display: none;
  }
}

@media screen and (max-width:768px) {
  .c2 {
    max-width: initial;
    height: 100vh;
    border-radius: 0;
    -webkit-transition: 400ms ease;
    transition: 400ms ease;
    padding: 20px;
    max-height: 100vh;
    overflow-y: auto;
  }
}

@media screen and (max-width:768px) {
  .c0[data-open=false] {
    opacity: 1;
    background-color: rgba(0,0,0,0);
  }
}

@media screen and (max-width:768px) {
  .c0[data-open=false] .c1 {
    opacity: 1;
    -webkit-transform: translateX(100vw);
    -ms-transform: translateX(100vw);
    transform: translateX(100vw);
  }
}

@media screen and (max-width:768px) {
  .c0 {
    padding: 0;
  }
}

<div
    class="c0"
    data-open="false"
  >
    <div
      class="c1 c2"
    >
      <img
        alt="close"
        class="c3"
        height="28"
        src="/icons/close-28x28.png"
        width="28"
      />
    </div>
  </div>
</DocumentFragment>
`;

exports[`should be open 1`] = `
<DocumentFragment>
  .c3 {
  position: absolute;
  right: 35px;
  top: 35px;
  cursor: pointer;
}

.c2 {
  width: 100%;
  max-width: 822px;
  background-color: white;
  border-radius: 25px;
  padding: 35px;
  position: relative;
  -webkit-transition: 400ms cubic-bezier(0.83,0,0.17,1);
  transition: 400ms cubic-bezier(0.83,0,0.17,1);
}

.c0 {
  width: 100vw;
  height: 100vh;
  z-index: 100000000;
  background-color: rgba(0,0,0,0.35);
  position: fixed;
  top: 0;
  left: 0;
  padding: 20px;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-transition: opacity 400ms ease,background-color 400ms;
  transition: opacity 400ms ease,background-color 400ms;
}

.c0[data-open=false] {
  pointer-events: none;
  opacity: 0;
}

.c0[data-open=false] .c1 {
  -webkit-transform: translateY(300px);
  -ms-transform: translateY(300px);
  transform: translateY(300px);
  opacity: 0;
}

@media screen and (max-width:768px) {
  .c3 {
    display: none;
  }
}

@media screen and (max-width:768px) {
  .c2 {
    max-width: initial;
    height: 100vh;
    border-radius: 0;
    -webkit-transition: 400ms ease;
    transition: 400ms ease;
    padding: 20px;
    max-height: 100vh;
    overflow-y: auto;
  }
}

@media screen and (max-width:768px) {
  .c0[data-open=false] {
    opacity: 1;
    background-color: rgba(0,0,0,0);
  }
}

@media screen and (max-width:768px) {
  .c0[data-open=false] .c1 {
    opacity: 1;
    -webkit-transform: translateX(100vw);
    -ms-transform: translateX(100vw);
    transform: translateX(100vw);
  }
}

@media screen and (max-width:768px) {
  .c0 {
    padding: 0;
  }
}

<div
    class="c0"
    data-open="true"
  >
    <div
      class="c1 c2"
    >
      <img
        alt="close"
        class="c3"
        height="28"
        src="/icons/close-28x28.png"
        width="28"
      />
    </div>
  </div>
</DocumentFragment>
`;
